configuration_go/abstr/kubernetes/thanos/receive/receive.go

```diff
@@ -0,0 +1,514 @@
+package receive
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+
+	cmdopt "github.com/observatorium/observatorium/configuration_go/abstr/kubernetes/cmdoption"
+	"github.com/observatorium/observatorium/configuration_go/k8sutil"
+	"github.com/observatorium/observatorium/configuration_go/schemas/thanos/log"
+	"github.com/observatorium/observatorium/configuration_go/schemas/thanos/option"
+	"github.com/observatorium/observatorium/configuration_go/schemas/thanos/reqlogging"
+	trclient "github.com/observatorium/observatorium/configuration_go/schemas/thanos/tracing/client"
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus/prometheus/model/relabel"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	dataVolumeName     string = "data"
+	defaultHTTPPort    int    = 10902
+	defaultGRPCPort    int    = 10901
+	defaultReceivePort int    = 19291
+	defaultNamespace   string = "observatorium"
+	defaultImage       string = "quay.io/thanos/thanos"
+)
+
+type GrpcCompressionType string
+type HashRingAlgorithm string
+
+const (
+	GrpcCompressionSnappy    GrpcCompressionType = "snappy"
+	GrpcCompressionNone      GrpcCompressionType = "none"
+	HashRingAlgorithmHashmod HashRingAlgorithm   = "hashmod"
+	HashRingAlgorithmKetama  HashRingAlgorithm   = "ketama"
+)
+
+// Label represents a single label configuration.
+type Label struct {
+	Key   string
+	Value string
+}
+
+func (l Label) String() string {
+	return l.Key + "=" + l.Value
+}
```

Comment:
Do we need to redefine this? We already use prometheus libs, so I think we can just use that instead?

Reply:
Yes we can. I just wanted to avoid the client to have to import prometheus to define this option. But I don't mind removing that. Do you confirm removal?

Reply:
Yup, I think importing prometheus for client using this lib would be ok.

Reply:
Ok, so actually this is trickier. 
When there is an option that can be repeated, it is set as a slice of the type for cmdopt, like `[]Label`. This way, it understands that the option must be repeated for each element. Then, each element of the slice must implement the `Stringer` interface. But in the prometheus labels case, it is defined on the slice type `Labels`:
```
type Labels []Label
func (ls Labels) String() string {
...
```

So I think we must keep it. I don't see another solution here. 😞 
I am reverting the commits I pushed for this.